File: src/DataScience/Julia/bridge.jl
make_device(name) = (
    power_on = () -> "$(name):on",
    mute = () -> "$(name):muted",
)

activate_basic(device) = device.power_on()
activate_mute(device) = device.mute()

tv = make_device("TV")
radio = make_device("Radio")

println("basic-tv=" * activate_basic(tv))
println("basic-radio=" * activate_basic(radio))
println("mute-tv=" * activate_mute(tv))
println("mute-radio=" * activate_mute(radio))

File: src/DataScience/MATLAB/bridge.m
function bridge
    tv = makeDevice("TV");
    radio = makeDevice("Radio");

    fprintf('basic-tv=%s\n', activateBasic(tv));
    fprintf('basic-radio=%s\n', activateBasic(radio));
    fprintf('mute-tv=%s\n', activateMute(tv));
    fprintf('mute-radio=%s\n', activateMute(radio));
end

function device = makeDevice(name)
    device.powerOn = @() name + ":on";
    device.mute = @() name + ":muted";
end

function value = activateBasic(device)
    value = device.powerOn();
end

function value = activateMute(device)
    value = device.mute();
end
